validator/sawtooth_validator/gossip/permission_verifier.py

```diff
@@ -241,6 +245,43 @@ def check_off_chain_transaction_roles(self, transactions):
 
         return True
 
+    def check_network_role(self, public_key):
+        """ Check the public key of a node on the network to see if they are
+            permitted to participate. The roles being checked are the
+            following, from first to last:
+                "network"
+                "default"
+
+            The first role that is set will be the one used to enforce if the
+            node is allowed.
+
+            Args:
+                public_key (string): The public key belonging to a node on the
+                    network
+        """
+        state_root = self._current_root_func()
+        if state_root == INIT_ROOT_KEY:
+            LOGGER.debug("Chain head is not set yet. Permit all.")
+            return True
+
+        identity_view = \
+            self._identity_view_factory.create_identity_view(state_root)
+
+        role = identity_view.get_role("network")
+
+        if role is None:
+            policy_name = "default"
+        else:
+            policy_name = role.policy_name
+        policy = identity_view.get_policy(policy_name)
+        if policy is not None:
+            if not self._allowed(public_key, policy):
+                LOGGER.debug(
+                    "Node is not permitted: %s.",
+                    public_key)
+                return False
+        return True
```

Comment:
Will there be a new method for every role that is created and needs to be checked? A more reusable function might be better. Something like `check_role()` that is generic.